src/test/java/uk/gov/pay/ledger/event/service/EventServiceTest.java

```diff
@@ -110,21 +110,39 @@ public void createIfDoesNotExistReturnsSuccessfulCreatedResponse() {
     public void createIfDoesNotExistReturnsSuccessfulIgnoredResponse() {
         when(mockEventDao.insertEventIfDoesNotExistWithResourceTypeId(event)).thenReturn(Optional.empty());
 
-        CreateEventResponse response = eventService.createIfDoesNotExist(event);
+        CreateEventResponse response = eventService.createOrUpdateIfExists(event);
 
         assertTrue(response.isSuccessful());
         assertThat(response.getState(), is(CreateEventResponse.CreateEventState.IGNORED));
     }
 
     @Test
-    public void createIfDoesNotExistReturnsNotSuccessfulResponse() {
+    public void createOrUpdateIfExistsReturnsNotSuccessfulResponse() {
         when(mockEventDao.insertEventIfDoesNotExistWithResourceTypeId(event))
                 .thenThrow(new RuntimeException("forced failure"));
 
-        CreateEventResponse response = eventService.createIfDoesNotExist(event);
+        CreateEventResponse response = eventService.createOrUpdateIfExists(event);
 
         assertFalse(response.isSuccessful());
         assertThat(response.getState(), is(CreateEventResponse.CreateEventState.ERROR));
         assertThat(response.getErrorMessage(), is("forced failure"));
     }
+
+    @Test
+    public void createORUpdateIfExistsUpdatesExistingEvent() {
+        Event existingEvent = EventFixture.anEventFixture()
+                .withResourceExternalId("an-event-id")
+                .toEntity();
+        Event newEvent = EventFixture.anEventFixture()
+                .withResourceExternalId(existingEvent.getResourceExternalId())
+                .withEventDate(existingEvent.getEventDate())
+                .withEventData("{\"key\": \"value\"}")
+                .toEntity();
+        when(mockEventDao.insertEventWithResourceTypeId(newEvent)).thenReturn(0L);
+        when(mockEventDao.updateIfExistsWithResourceTypeId(newEvent)).thenReturn(Optional.of(1L));
+        eventService.createOrUpdateIfExists(existingEvent);
+        CreateEventResponse response = eventService.createOrUpdateIfExists(newEvent);
```

Comment:
Don't think you need the 2 events and to call the method twice. As you've mocked out the dao it doesn't actually do anything. You can just use the `event` field rather than making events for this test and only call `createOrUpdateIfExists` once.